test/input.for.test/preprocessing_pdf.py

```diff
@@ -76,92 +85,15 @@ def preprocess_pdf(abstract, features, tfidf, num_sentences=5):
 
             if ans =='A' and len(top_term) > 0:
                 del top_term[0]
-            #updated here
-            out_str += "{}￨{}￨{}￨{}￨{} ".format(tok_text,low,token['pos'],token['ner'],ans)
 
-        if '￨A' in out_str:
-            print(out_str)
+            out_str += "{}￨{}￨{}￨{}￨{} ".format(tok_text, low, token['pos'],
+                                                   token['ner'], ans)
 
+        if '￨A' in out_str:
+            output_vec.append(out_str)
```

Comment:
Will append create a list? We just need a new string on each line but not in the form of a list with []. Also we need to make sure there are no spaces at the end of each line.

Reply:
So, `output_vec` is a list and the `.append` will append each string as an item in the list. Since the `out_str += "{}￨{}￨{}￨{}￨{} ".format(...)` puts a space at the end of each string added to `out_str`, I've modified the code to be the following: `output_vec.append(out_str[:-1])`, which will strip the last whitespace from each string added to `output_vec`. Later, the `'\n'.join(output_vec)` 'flattens the list to just be strings separated by a carriage return. This will structure a return object akin to what you've produced.